src/dagmc/dagmcmetadata.hpp

```diff
@@ -8,118 +8,316 @@
+   *                value after removing the key and delimiter. If false, it
+   *                will return the property value as is. Default is true.
+   *
+   * @return Returns the value of the specified key from the property string.
+   */
   std::string return_property(std::string property_string, std::string property,
                               std::string delimiter = ":", bool chopped = true);
 
-  // test to see if string is an int
+  /**
+   * @brief Tests if a string can be converted to an integer.
+   *
+   * @param value The string to be tested.
+   *
+   * @return Returns true if the string can be converted to an integer,
+   *         false otherwise.
+   */
   bool try_to_make_int(std::string value);
 
   // private member functions
  private:
-  // parse the material data
+  /**
+   * @brief Parses the material data from the associated DagMC instance.
+   */
   void parse_material_data();
-  // parse the importance data
+
+  /**
+   * @brief Parses the importance data from the associated DagMC instance.
+   */
   void parse_importance_data();
-  // parse the boundary data
+
+  /**
+   * @brief Parses the boundary data from the associated DagMC instance.
+   */
   void parse_boundary_data();
-  // parse the tally data
+
+  /**
+   * @brief Parses the tally surface data from the associated DagMC instance.
+   */
   void parse_tally_surface_data();
-  // parse the tally data
+
+  /**
+   * @brief Parses the tally volume data from the associated DagMC instance.
+   */
   void parse_tally_volume_data();
-  // finalise the count data
+
+  /**
+   * @brief Finalizes the counters after all data has been parsed.
+   */
   void finalise_counters();
 
-  // Parse property for entities with the specified dimension and delimiters.
-  // Optionally remove duplicate property values if necessary.
+  /**
+   * @brief Parses property for entities with the specified dimension and
+   *        delimiters. Optionally removes duplicate property values if
+   * necessary.
+   *
+   * @param property The name of the property to retrieve.
+   * @param dimension The dimension of the entities for which to retrieve the
+   *                  property.
+   * @param delimiters The delimiters used to separate the data in the string.
+   * @param remove_duplicates Optional. If true, the function will remove
+   *                          duplicate property values. Default is true.
+   *
+   * @return Returns a map where the keys are the entity handles and the values
+   *         are vectors of property values.
+   */
   std::map<moab::EntityHandle, std::vector<std::string>>
   get_property_assignments(std::string property, int dimension,
                            std::string delimiters,
                            bool remove_duplicates = true);
 
-  // remove duplicate properties from the vector of properties
+  /**
+   * @brief Removes duplicate properties from a vector of properties.
```

Comment:
Is it possible to give a simple example of how this might manifest itself? Like the neutron example you gave below for the `set_remove_rich()` method?

Reply:
✅